metadata/CMakeLists.txt

```diff
@@ -17,12 +17,13 @@
 add_library(hyperwarp-metadata SHARED metadata.c)
 add_custom_command(TARGET hyperwarp-metadata PRE_BUILD COMMAND astyle ARGS "${CMAKE_CURRENT_SOURCE_DIR}/*.c,*.h")
 
-target_link_libraries(hyperwarp-metadata hyperwarp-metadata-model uuid)
+target_link_libraries(hyperwarp-metadata hyperwarp-metadata-model uuid dl)
 
 target_include_directories(hyperwarp-metadata PUBLIC
                           "${PROJECT_BINARY_DIR}"
                           "${PROJECT_SOURCE_DIR}"
                           "${CMAKE_CURRENT_SOURCE_DIR}"
+                          "${GLIB2_MODULE_INCLUDE_DIRS}"
```

Comment:
seems you are not using glib2 anymore so I guess this can be removed

Reply:
indeed, as discussed, I've switched to pure dlopen() as for right now it works for us equally both.